### DISCUSSION TITLE
Nuxt.js on Vercel: Serverless Function exceeds 50mb

### DISCUSSION THREAD

**Discussion by jannisstefanis** [vercel/vercel#8753] [Help] [answered]
Hey There!

About a month ago, I deployed my SSR Nuxt Application to Vercel. It worked great, new changes were deployed automatically and I haven't encountered any problems. 
Yesterday I made some changes, but the build is failing since then. It gives me the following error:

`The Serverless Function "index" is 67.06mb which exceeds the maximum size limit of 50mb.`

When looking that error up, some people suggested to check the bundle size, but the library I added in the latest commit (vuedraggable) is roughly around 500kb...
I tried to move all my dependencies into devDependencies (as suggested here: https://github.com/nuxt/vercel-builder/issues/284), which unfortunately hasn't worked.
I even tried to revert my git changes to the latest working commit but also this build has been failing with the same error.

This is my package.json: 
```
{
  "name": "irl",
  "version": "1.0.0",
  "private": true,
  "scripts": {
    "dev": "nuxt",
    "build": "nuxt build",
    "start": "nuxt start",
    "generate": "nuxt generate"
  },
  "devDependencies": {
    "@nuxtjs/auth-next": "^5.0.0-1648802546.c9880dc",
    "@nuxtjs/axios": "^5.13.6",
    "@nuxtjs/recaptcha": "^1.1.1",
    "@yeger/vue2-masonry-wall": "^2.0.49",
    "core-js": "^3.19.3",
    "nuxt": "^2.15.8",
    "vue": "^2.6.14",
    "vue-gallery": "^2.0.1",
    "vue-lazy-hydration": "^2.0.0-beta.4",
    "vue-server-renderer": "^2.6.14",
    "vue-smooth-dnd": "^0.8.1",
    "vue-template-compiler": "^2.6.14",
    "vue-underscore": "^0.1.4",
    "vuedraggable": "^2.24.3",
    "webpack": "^4.46.0",
    "@fortawesome/free-brands-svg-icons": "^6.1.2",
    "@fortawesome/free-solid-svg-icons": "^6.2.0",
    "@nuxtjs/fontawesome": "^1.1.2",
    "@nuxtjs/google-fonts": "^3.0.0-0",
    "@nuxtjs/tailwindcss": "^4.2.1",
    "postcss": "^8.4.4"
  }
}
```

What can I do?
Thanks in Advance!

**Comment by amyegan** [comment, marked as answer]
Hi @jannisstefanis! It looks like your packages [allow automatic minor revision updates](https://docs.npmjs.com/about-semantic-versioning#using-semantic-versioning-to-specify-update-types-your-package-can-accept). The `^` means that any minor release from the specified version or later can be used. It's possible that newer (larger) versions of some packages are being used with new builds than were used in previous builds.

I don't know which of these packages are being used in your Serverless Function, but you can use [bundlejs.com](https://bundlejs.com) or [bundlephobia.com](https://bundlephobia.com/) to check which packages might be causing the problem.

The Serverless Function size might not be exclusively caused by installed packages. If you're [using files in your Serverless Function](https://vercel.com/guides/how-can-i-use-files-in-serverless-functions), that could also cause the size limit to be exceeded.
